mrc/localization/locator.py

```diff
@@ -1,36 +1,41 @@
 import threading
-from dynaconf import settings
-import itertools
 import time
+from typing import Callable
 
 
 class Locator(threading.Thread):
-    def __init__(self):
+    def __init__(self, read_data: Callable, process_data: Callable, interval: float):
```

Comment:
How about default values for process_data( None ), and interval( 0 )? process_data do not have to be used and interval probably will be 0 most of time.